### ISSUE TITLE
 Save operation is not supported by sqlite driver

### ISSUE THREAD

**Issue by ltanme** [gogf/gf#2473]
### 1. What version of `Go` and system type/arch are you using?
windows 10 x86 amd ryzen7

go 1.17.12

### 2. What version of `GoFrame` are you using?

GF Version:  v2.1.2

### 3. Can this issue be re-produced with the latest release?
no

### 4. What did you do?

GF Version:  v2.1.2

### 5. What did you expect to see?

func (a *cSetting) Save(ctx context.Context, req *v1.SettingSaveReq) (res *v1.SettingSaveRes, err error) {
	// g.Log().Info(ctx, req)
	// service.Setting().Set(ctx, req.K, string(req.V))
	// aa := service.Setting().Set(ctx, string(req.K), string(req.V))
	// fmt.Println(aa)

	// service.Setting().GetVar("homepageUrl")

	r, err := dao.Setting.SettingDao.DB().Save(ctx, "gf_setting", gdb.Map{
		"k": "1111",
		"v": "3333",
	})

	fmt.Println(r)

	// v, err := dao.Setting.Ctx(ctx).Data(entity.Setting{
	// 	K: "HomepageUrl",
	// 	V: "33333333333333333333",
	// }).Save()

	g.Log().Info(ctx, req.K, req.V)
	// v, err := service.Setting().GetVar(ctx, "abcd")

	g.Log().Info(ctx, "查到v值", 0)
	return
}

### 6. What did you see instead?
报错如下

2023-02-24 18:28:48.576 [DEBU] {bc2fafd8eabb46170803bb7797f66466} [  1 ms] [default] [rows:4  ] PRAGMA TABLE_INFO(gf_setting)
<nil>
2023-02-24 18:28:48.578 [INFO] {bc2fafd8eabb46170803bb7797f66466} homepageUrl 222222
2023-02-24 18:28:48.579 [INFO] {bc2fafd8eabb46170803bb7797f66466} 查到v值 0
2023-02-24 18:28:48.580 {bc2fafd8eabb46170803bb7797f66466} 200 "POST http localhost:8199 /setting/save HTTP/1.1" 0.006, 4.2.2.2, "http://localhost:8199/setting/index", "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/110.0.0.0 Safari/537.36 Edg/110.0.1587.50", 59, "Not Supported", ""
Stack:
1. Save operation is not supported by sqlite driver
   1).  github.com/gogf/gf/contrib/drivers/sqlite/v2.(*Driver).DoInsert
        E:/GO/pkg/mod/github.com/gogf/gf/contrib/drivers/sqlite/v2@v2.2.6/sqlite.go:179
   2).  github.com/gogf/gf/v2/database/gdb.(*HookInsertInput).Next
        E:/GO/pkg/mod/github.com/gogf/gf/v2@v2.2.1/database/gdb/gdb_model_hook.go:122
   3).  github.com/gogf/gf/v2/database/gdb.(*Model).doInsertWithOption
        E:/GO/pkg/mod/github.com/gogf/gf/v2@v2.2.1/database/gdb/gdb_model_insert.go:366
   4).  github.com/gogf/gf/v2/database/gdb.(*Model).Save
        E:/GO/pkg/mod/github.com/gogf/gf/v2@v2.2.1/database/gdb/gdb_model_insert.go:237
   5).  github.com/gogf/gf/v2/database/gdb.(*Core).Save
        E:/GO/pkg/mod/github.com/gogf/gf/v2@v2.2.1/database/gdb/gdb_core.go:392
   6).  focus-single/internal/controller.(*cSetting).Save
        E:/aa/go-demo/focus-single/internal/controller/setting.go:38
   7).  github.com/gogf/gf/v2/net/ghttp.(*middleware).callHandlerFunc.func1
        E:/GO/pkg/mod/github.com/gogf/gf/v2@v2.2.1/net/ghttp/ghttp_request_middleware.go:152        
   8).  github.com/gogf/gf/v2/net/ghttp.niceCallFunc
        E:/GO/pkg/mod/github.com/gogf/gf/v2@v2.2.1/net/ghttp/ghttp_func.go:55
   9).  github.com/gogf/gf/v2/net/ghttp.(*middleware).callHandlerFunc
        E:/GO/pkg/mod/github.com/gogf/gf/v2@v2.2.1/net/ghttp/ghttp_request_middleware.go:129        
   10). github.com/gogf/gf/v2/net/ghttp.(*middleware).Next.func1
        E:/GO/pkg/mod/github.com/gogf/gf/v2@v2.2.1/net/ghttp/ghttp_request_middleware.go:75
   11). github.com/gogf/gf/v2/util/gutil.TryCatch
        E:/GO/pkg/mod/github.com/gogf/gf/v2@v2.2.1/util/gutil/gutil.go:56
   12). github.com/gogf/gf/v2/net/ghttp.(*middleware).Next
        E:/GO/pkg/mod/github.com/gogf/gf/v2@v2.2.1/net/ghttp/ghttp_request_middleware.go:49
   13). focus-single/internal/logic/middleware.(*sMiddleware).Auth
        E:/aa/go-demo/focus-single/internal/logic/middleware/middleware.go:108
   14). github.com/gogf/gf/v2/net/ghttp.(*middleware).Next.func1.1
        E:/GO/pkg/mod/github.com/gogf/gf/v2@v2.2.1/net/ghttp/ghttp_request_middleware.go:55
   15). github.com/gogf/gf/v2/net/ghttp.(*middleware).Next.func1
        E:/GO/pkg/mod/github.com/gogf/gf/v2@v2.2.1/net/ghttp/ghttp_request_middleware.go:54
   16). focus-single/internal/logic/middleware.(*sMiddleware).ResponseHandler
        E:/aa/go-demo/focus-single/internal/logic/middleware/middleware.go:31
   17). focus-single/internal/logic/middleware.(*sMiddleware).Ctx
        E:/aa/go-demo/focus-single/internal/logic/middleware/middleware.go:86
   18). github.com/gogf/gf/v2/net/ghttp.internalMiddlewareServerTracing
        E:/GO/pkg/mod/github.com/gogf/gf/v2@v2.2.1/net/ghttp/ghttp_middleware_tracing.go:78
   19). github.com/gogf/gf/v2/net/ghttp.(*middleware).Next.func1.5
        E:/GO/pkg/mod/github.com/gogf/gf/v2@v2.2.1/net/ghttp/ghttp_request_middleware.go:96_middleware.go:96
   20). github.com/gogf/gf/v2/net/ghttp.(*middleware).Next.func1          _middleware.go:95
        E:/GO/pkg/mod/github.com/gogf/gf/v2@v2.2.1/net/ghttp/ghttp_request_middleware.go:95                                                         handler.go:121
   21). github.com/gogf/gf/v2/net/ghttp.(*Server).ServeHTTP
        E:/GO/pkg/mod/github.com/gogf/gf/v2@v2.2.1/net/ghttp/ghttp_server_handler.go:121

**Comment by gqcn** [comment]
@ltanme Hi, the `Save` function is not supported by `sqlite`. But can be implemented through custom driver implements https://github.com/gogf/gf/tree/master/contrib/drivers/sqlite Any contribution on this will be appreciated.

**Comment by ltanme** [comment]
this code  "service.Setting().GetVar(ctx, "abcd") "  is not supported???
 why?

**Comment by gqcn** [comment]
> this code "service.Setting().GetVar(ctx, "abcd") " is not supported???
> why?


Not really?

**Comment by ltanme** [comment]
At first I thought I could use the following sentence to save data to the gf_setting table: service.Setting().Set(ctx, req.K, string(req.V)) But later I found that it didn’t work and there was no data in the table. 
I want to add a controller file for gf_setting to perform CRUD operations on this table. 
So I added a controller file for gf_setting. Added a /save interface to save the key value fields of the setting table. 
Below I wrote this in the controller file, but got an error: 

r, err := dao.Setting.SettingDao.DB().Save(ctx, "gf_setting", gdb.Map{ "k": "1111", "v": "3333", }) 
Reported: .
 Save operation is not supported by sqlite driver.
